qa_eval/answer_evaluation.py

```diff
@@ -0,0 +1,55 @@
+import csv
+import sys
+from pathlib import Path
+
+from openai import OpenAI
+from tqdm import tqdm
+
+
+DATA_FILE_PATH = '../data/paws/knowledge-hub.tsv'
+PROMPT_FILE_PATH = 'prompts/template.md'
+OUT_FILE_PATH = 'results/knowledge-hub.tsv'
+OUT_FIELDS = ['#T', '#P', '#TP', 'LLM reasoning']
+LLM_MODEL = 'gpt-4o-mini'
+TEMPERATURE = 0.0
+
+
+def call_llm(client, prompt) -> str:
+    try:
+        response = client.chat.completions.create(
+            model=LLM_MODEL,
+            messages=[{'role': 'user', 'content': prompt}],
+            temperature=TEMPERATURE
+        )
+        return response.choices[0].message.content.strip('\n')
+    except Exception as e:
+        return str(e).replace('\n', '    ')
+
+
+def evaluate_answers():
+    client = OpenAI()
+    with open(PROMPT_FILE_PATH, 'r', encoding='utf-8') as f:
+        prompt_template = f.read()
+    with open(DATA_FILE_PATH, encoding='utf-8') as in_f:
+        reader = csv.DictReader(in_f, delimiter='\t')
+        Path(OUT_FILE_PATH).parent.mkdir(parents=True, exist_ok=True)
+        print(f'Writing results to {OUT_FILE_PATH}')
+        with open(OUT_FILE_PATH, 'w', encoding='utf-8') as out_f:
+            out_f.write('\t'.join(OUT_FIELDS) + '\n')
+            rows = [row for row in reader]
+            for row in tqdm(rows):
+                prompt = prompt_template.format(
+                    question=row['Question'],
+                    reference_answer=row['Reference answer'],
+                    candidate_answer=row['Actual answer'],
+                )
+                llm_output = call_llm(client, prompt)
+                out_f.write(llm_output + '\n')
+                out_f.flush()
+                print('.', end='')
+                sys.stdout.flush()
+        print()
+
+
+if __name__ == '__main__':
+    evaluate_answers()
```

Comment:
I think it would be good to document how the user can use this script in the README section `Usage`. Also how about defining this as in the pyproject.toml project.scripts?